kube/src/client/middleware/base_uri.rs

```diff
@@ -0,0 +1,109 @@
+//! Set base URI of requests.
+use http::{uri, Request};
+use tower::{Layer, Service};
+
+/// Layer that applies [`SetBaseUri`] which makes all requests relative to the base URI.
+///
+/// Path in `base_uri` is preseved.
+#[derive(Debug, Clone)]
+pub struct SetBaseUriLayer {
```

Comment:
Think this one should just be named `BaseUriLayer` for consistency. Our core exported layers are all without verbs (except for the temporary `AddAuthorizationLayer` taken from the linked tower-http pr), and having that consistency - where possible - feels cleaner to me.

NB: We might not be able to go for a verb-less style in general / long-term, as [tower-http](https://docs.rs/tower-http/0.1.0/tower_http/) does uncover legit cases where they need more than one verb on the same resource (`SetResponseHeaderLayer` vs `PropagateHeaderLayer`), but they also drop the verb when there's no disambiguation needed.